src/main/kotlin/com/rstudio/shinycannon/Main.kt

```diff
@@ -290,11 +325,11 @@ class EnduranceTest(val argsStr: String,
     fun run() {
         val rec = readRecording(recording, logger)
         val connectApiKeyRequired = rec.props.rscApiKeyRequired
-        if (connectApiKeyProvided && !connectApiKeyRequired) {
+        if (creds.hasConnectApiKey() && !connectApiKeyRequired) {
             logger.error(("Unexpected RStudio Connect API key provided for playback. Recording was made without an API key."))
             exitProcess(1)
```

Comment:
`exitProcess(1)` seems a little harsh here (and in the next conditional too). Can we just log this as a warning and continue? I'm thinking of the case where authentication was added (or removed) since the recording was made.

Reply:
Playback does not work as expected.  It behaves more as an error than a warning.

If auth is used, the websocket messages received are different than the websocket messages without auth. In both of these `on/off` or `off/on` situations occur, the playback fails to execute.

Reply:
OK. We can definitely fix that in the future if people want it, just need to massage that `custom` websocket message when it comes back so it's normalized.